### PR TITLE
RDRP-471: QA output for total FTEs

### PR DESCRIPTION
**Pull request by coatet**
## Pull Request submission

- Add function to get counts of FTE broken down by totals, researchers, technicians and other
- Add function to estimation QA outputs

When testing, remember to turn `output_estimation_qa` on in the config, caught me out for a while! Also open to other ways to arrange the code, since it's not like the other estimation QA outputs. 

#### Closes or fixes

Closes #124 

#### Code

- [ ] **Code runs** The code runs on my machine and/or CDSW
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `environment.yaml`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

**Review by AnneONS — COMMENTED** [review]
Looks good! Really neat code.

Just wondering whether it might be good to actually do the output in the output module, where all the other outputs go... I take your point about it being a qa output so a bit different, but Jen has now added another qa output to the outputs module. It just might be a bit easier for someone to find.

File: src/outputs/total_fte.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        write_csv (Callable): Function to write to a csv file.
+         This will be the hdfs or network version depending on settings.
+        run_id (int): The current run id
+
+    """
+    NETWORK_OR_HDFS = config["global"]["network_or_hdfs"]
+    est_path = config[f"{NETWORK_OR_HDFS}_paths"]["estimation_path"]
+
+    imp_marker_list = ["R", "TMI", "CR", "MoR"]
+    sliced_df = df.copy().loc[df["imp_marker"].isin(imp_marker_list)]
```
Comment (AnneONS) [inline src/outputs/total_fte.py]: Actually, there's some code in output_main.py that does this already...

File: src/outputs/total_fte.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    totals_names = ["Total",
+                    "Scientists and engineers",
+                    "Technicians, laboratory assistants and draughtsmen",
+                    "Administrative, clerical, industrial and other staff"]
```
Comment (AnneONS) [inline src/outputs/total_fte.py:33]: This is one of those cases where I wonder whether the requirements that Nick Powell wrote are actually needed, or whether  the business area would be happy with the actual column names we're using?

Comment (zorge69) [reply src/outputs/total_fte.py:33]: @AnneONS let's use the names Nick suggested. The BaU might have to do less work renaming our techy acronyms to user-readable ones.

Comment (AnneONS) [reply src/outputs/total_fte.py:33]: @zorge69 I've asked in the Stand up chat, Dan says he'd prefer the new names.

**Review by zorge69 — APPROVED** [review]
The code works, creates an output.
The numbers are rounded to 4 decimal places - ok for testing, needs to be rounded for publication,
Runs very quickly.
Not sure is export from DAP will be required

**Pull request merged**